src/analysis/normed_space/linear_isometry.lean

```diff
@@ -13,23 +13,27 @@ embedding of `E` into `F` and `linear_isometry_equiv` (notation: `E ≃ₗᵢ[R]
 isometric equivalence between `E` and `F`.
 
 We also prove some trivial lemmas and provide convenience constructors.
+
+Since a lot of elementary properties don't require `∥x∥ = 0 → x = 0` we start setting up the
+theory for `semi_normed_space` and we specialize to `normed_space` when needed.
 -/
 open function set
 
-variables {R E F G G' : Type*} [semiring R]
-  [normed_group E] [normed_group F] [normed_group G] [normed_group G']
+variables {R E F G G' E₁ F₁ : Type*} [semiring R]
+  [semi_normed_group E] [semi_normed_group F] [semi_normed_group G] [semi_normed_group G']
   [semimodule R E] [semimodule R F] [semimodule R G] [semimodule R G']
+  [normed_group E₁] [normed_group F₁] [semimodule R E₁] [semimodule R F₁]
 
 /-- An `R`-linear isometric embedding of one normed `R`-module into another. -/
-structure linear_isometry (R E F : Type*) [semiring R] [normed_group E] [normed_group F]
-  [semimodule R E] [semimodule R F] extends E →ₗ[R] F :=
+structure linear_isometry (R E F : Type*) [semiring R] [semi_normed_group E]
+  [semi_normed_group F] [semimodule R E] [semimodule R F] extends E →ₗ[R] F :=
 (norm_map' : ∀ x, ∥to_linear_map x∥ = ∥x∥)
 
 notation E ` →ₗᵢ[`:25 R:25 `] `:0 F:0 := linear_isometry R E F
 
 namespace linear_isometry
 
-variables (f : E →ₗᵢ[R] F)
+variables (f : E →ₗᵢ[R] F) (f₁ : E₁ →ₗᵢ[R] F)
```

Comment:
Please add a comment saying that `f` is defined on a `semi_normed_group` and `f₁` is defined on a `normed_group`.